Kickstarter-iOS/Views/Controllers/CommentsViewController.swift

```diff
@@ -8,23 +8,80 @@ import UIKit
 internal final class CommentsViewController: UITableViewController {
   // MARK: - Properties
 
-  fileprivate let viewModel: CommentsViewModelType = CommentsViewModel()
+  private let viewModel: CommentsViewModelType = CommentsViewModel()
+  private let dataSource = CommentsDataSource()
+  private let refreshIndicator = UIRefreshControl()
+
+  // MARK: - Configuration
+
+  internal static func configuredWith(project: Project? = nil, update: Update? = nil) -> CommentsViewController {
+    let vc = CommentsViewController.instantiate()
+    vc.viewModel.inputs.configureWith(project: project, update: update)
+
+    return vc
+  }
 
   // MARK: - Lifecycle
 
   internal override func viewDidLoad() {
     super.viewDidLoad()
-
+    
+    self.setupTableView()
+    self.setupRefreshControl()
+    
     self.viewModel.inputs.viewDidLoad()
   }
 
   // MARK: - Styles
 
   internal override func bindStyles() {
     super.bindStyles()
+    // TODO: Fill this out with comment card, comment composer.
   }
 
   // MARK: - View Model
 
-  internal override func bindViewModel() {}
+  internal override func bindViewModel() {
+    self.viewModel.outputs.loadCommentsIntoDataSource
+      .observeForUI()
+      .observeValues { comments in
+        self.dataSource.updateCommentsSection(comments: comments)
+        self.tableView.reloadData()
+      }
+    
+    self.viewModel.outputs.commentsAreLoading
+      .observeForUI()
+      .observeValues { [weak self] in
+        $0 ? self?.refreshControl?.beginRefreshing() : self?.refreshControl?.endRefreshing()
+      }
+  }
+  
+  // MARK: Helpers
```

Comment:
While these sorts of functions are fine, I've found grouping things under the name of _helpers_ usually ends up having almost everything under it 😄 Kind've like our `Library` framework 🤦‍♂️.

Could you structure this more like some of our other VC's? Just for convention's sake, take a look at https://github.com/kickstarter/ios-oss/blob/master/Kickstarter-iOS/Views/Controllers/EditorialProjectsViewController.swift perhaps.

Reply:
I'm not sure where to put the `self.tableView` setup code so I checked a few other table view controllers and it seems to be done in `viewDidLoad`, so I just moved it there.